src/applications/move_reminder.c

```diff
@@ -0,0 +1,1708 @@
+        StringTemplate_SetNickname(controller->stringTemplate, 0, Pokemon_GetBoxPokemon(controller->data->mon));
+        StringTemplate_SetMoveName(controller->stringTemplate, 1, MoveReminder_GetSelectedMove(controller));
+        break;
+
+    case MOVE_REMINDER_STR_1_2_AND_POOF:
+        StringTemplate_SetNickname(controller->stringTemplate, 0, Pokemon_GetBoxPokemon(controller->data->mon));
+        StringTemplate_SetMoveName(controller->stringTemplate, 1, MoveReminder_GetForgottenMove(controller));
+        break;
+
+    case MOVE_REMINDER_STR_LEARNED_MOVE_FANFARE:
+        StringTemplate_SetNickname(controller->stringTemplate, 0, Pokemon_GetBoxPokemon(controller->data->mon));
+        StringTemplate_SetMoveName(controller->stringTemplate, 1, MoveReminder_GetSelectedMove(controller));
+        break;
+
+    case MOVE_REMINDER_STR_ASK_STOP_TRYING_TO_TEACH_MOVE:
+        StringTemplate_SetMoveName(controller->stringTemplate, 1, MoveReminder_GetSelectedMove(controller));
+        break;
+
+    case MOVE_REMINDER_STR_MON_DID_NOT_LEARN_MOVE:
+        StringTemplate_SetNickname(controller->stringTemplate, 0, Pokemon_GetBoxPokemon(controller->data->mon));
+        StringTemplate_SetMoveName(controller->stringTemplate, 1, MoveReminder_GetSelectedMove(controller));
+        break;
+
+    case MOVE_REMINDER_STR_PAYMENT:
+        StringTemplate_SetPlayerName(controller->stringTemplate, 2, controller->data->trainerInfo);
+        break;
+
+    case MOVE_REMINDER_STR_ASK_FORGET_THIS_MOVE:
+        StringTemplate_SetMoveName(controller->stringTemplate, 0, MoveReminder_GetForgottenMove(controller));
+        break;
+    }
+
+    Strbuf *strbuf = MessageLoader_GetNewStrbuf(controller->messageLoader, sMessageEntryIDs[controller->data->isMoveTutor][str]);
+    StringTemplate_Format(controller->stringTemplate, controller->strbuf, strbuf);
+    Strbuf_Free(strbuf);
+}
+
+static u16 MoveReminder_GetSelectedMove(MoveReminderController *controller)
+{
+    return controller->data->moves[controller->data->listPos + controller->data->cursorPos];
+}
+
+static u16 MoveReminder_GetForgottenMove(MoveReminderController *controller)
+{
+    return (u16)Pokemon_GetValue(controller->data->mon, MON_DATA_MOVE1 + controller->data->moveSlot, NULL);
+}
+
+static void MoveReminder_SetMessageBoxText(MoveReminderController *controller, u32 str)
+{
+    Window_FillTilemap(&controller->windows[MOVE_REMINDER_WIN_MESSAGE_BOX], 15);
+    MoveReminder_SetStringTemplate(controller, str);
+    RenderControlFlags_SetCanABSpeedUpPrint(TRUE);
+    RenderControlFlags_SetAutoScrollFlags(AUTO_SCROLL_DISABLED);
+
+    controller->textPrinterID = Text_AddPrinterWithParams(&controller->windows[MOVE_REMINDER_WIN_MESSAGE_BOX], FONT_MESSAGE, controller->strbuf, 0, 0, Options_TextFrameDelay(controller->data->options), MoveReminder_TextPrinterCallback);
+}
+
+static BOOL MoveReminder_TextPrinterCallback(TextPrinterTemplate *printer, u16 param1)
+{
+    switch (param1) {
```

Comment:
Suggestion: identify this parameter

Reply:
Params in text printer callbacks are still undocumented at the moment - tracking them down and finding out what they do is out of scope of this PR.